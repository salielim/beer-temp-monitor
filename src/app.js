import React from 'react';
import ReactDOM from 'react-dom';
import { Table } from './components/table';

export class App extends React.Component {
  reloadPage() {
    window.location.reload();
  }

  render() {
    return (
      <div>
        <h1>Beer Temperature Monitoring</h1>
        <button
          className="btn btn-outline-dark btn-lg btn-block"
          onClick={() => this.reloadPage()}
        >
          Click here for live temperature updates
        </button>
        <Table />
      </div>
    );
  }
}