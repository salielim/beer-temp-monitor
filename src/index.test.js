import React from 'react';
import renderer from 'react-test-renderer';
import { App } from './app';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
