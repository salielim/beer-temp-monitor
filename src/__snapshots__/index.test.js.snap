// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <h1>
    Beer Temperature Monitoring
  </h1>
  <button
    className="btn btn-outline-dark btn-lg btn-block"
    onClick={[Function]}
  >
    Click here for live temperature updates
  </button>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          ID
        </th>
        <th
          scope="col"
        >
          Beer Type
        </th>
        <th
          scope="col"
        >
          Min Temp
        </th>
        <th
          scope="col"
        >
          Max Temp
        </th>
        <th
          scope="col"
        >
          Current Temp
        </th>
        <th
          scope="col"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          scope="row"
        >
          1
        </td>
        <td>
          Pilsner
        </td>
        <td>
          4
        </td>
        <td>
          6
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
      <tr>
        <td
          scope="row"
        >
          2
        </td>
        <td>
          IPA
        </td>
        <td>
          5
        </td>
        <td>
          6
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
      <tr>
        <td
          scope="row"
        >
          3
        </td>
        <td>
          Lager
        </td>
        <td>
          4
        </td>
        <td>
          7
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
      <tr>
        <td
          scope="row"
        >
          4
        </td>
        <td>
          Stout
        </td>
        <td>
          6
        </td>
        <td>
          8
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
      <tr>
        <td
          scope="row"
        >
          5
        </td>
        <td>
          Wheat beer
        </td>
        <td>
          3
        </td>
        <td>
          5
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
      <tr>
        <td
          scope="row"
        >
          6
        </td>
        <td>
          Pale Ale
        </td>
        <td>
          4
        </td>
        <td>
          6
        </td>
        <td />
        <td>
          Ok
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
